<!-- paper link: https://www.nature.com/articles/nrg3871 -->
### 1. Why is important to construct a CNV map on health individuals of various ethnicities?
>Clinical geneticists need to dis- criminate pathogenic or high-risk variants from benign variants in patients, and they do so with information about CNVs found in apparently healthy individu- als, which can include parents, siblings and popula- tion controls.

### 2. What is the CNV size that the authors defined?
50 bp to 3Mb


### 3. What are the primary approaches used for CNV detection? And what are the advantages and limitations of these technologies?
Microarrays and next-generation sequencing (NGS)

>NGS, call smaller variants and are biased towards the detection of deletions16,43. Sequencing-based approaches have both high sensitivity and the ability to provide accurate sequence-level breakpoint resolution, whereas array-based approaches have a limited resolu- tion capacity.


### 4. The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering?
The criteria for cluster filtering are the number of distinct subjects that carry the variant and the number of distinct studies with at least one variant in the cluster.

>The filter based on the number of subjects excluded singletons, and variants that are supported by a larger number of subjects are less likely to be false positives. The filter based on the number of studies ensured the exclu- sion of potential study-specific artefacts. 

### 5. What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively?
#### stringency level 1
at least one subject and one study (every variant, including singletons)
#### stringency level 2 - inclusive map 
at least two subjects and one study
#### stringency level 12 - stringent map
at least two subjects and two studies

### 6. Which percentage of the genome contributes to CNV in inclusive map and stringent map respectively?
inclusive map: 9.5%

stringent map: 4.8%


### 7. By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their findings in this paper?
Non-coding genes are more variable. Varriations on protein-coding genes do more often lead to disease or death and therefore are under negative selection.


### 8. The authors generated a null CNV map and found genes for which at least 85% of the exons were homozygous deleted. What are the functions of these genes? And why did the authors say that they seem to be non-essential?
>(Homozygous deleted genes)
>The function of non-essential genes may be redundant, or they may be related to late-onset pheno- typic consequences that do not substantially reduce the fitness. 

These genes seem to be non-essential, because they are missing in many apparently healthy individuals. If the genes were essential, the individuals who lack these genes would have to be affected in some way.


### 9. If you are a medical doctor, how do you use this map as a tool to assess the clinical importance of a CNV?
>(Clinical application of the CNV map part in Discussion)

