<!-- paper link: https://www.nature.com/articles/nrg3871 -->
### Why is important to construct a CNV map on health individuals of various ethnicities?
>(Introduction)


### What is the CNV size that the authors defined?
>(Box 1 mentioned in introduction)
Noted: The CNV size definition is still under debate and may be different in other literature



### What are the primary approaches used for CNV detection? And what are the advantages and limitations of these technologies?
>(CNV discoveries)



### The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering?
>(The CNV map)



### What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively?
>(The CNV map)



### Which percentage of the genome contributes to CNV in inclusive map and stringent map respectively?
>(Properties of the CNV map)



### By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their findings in this paper?
>(Functional impact of CNV)



### The authors generated a null CNV map and found genes for which at least 85% of the exons were homozygous deleted. What are the functions of these genes? And why did the authors say that they seem to be non-essential?
>(Homozygous deleted genes)



### If you are a medical doctor, how do you use this map as a tool to assess the clinical importance of a CNV?
>(Clinical application of the CNV map part in Discussion)

