## 1. Why is important to construct a CNV map on health individuals of various ethnicities?
To distinct between high-risk or even pathological variants and benign variants.As CNVs have different frquencies of occurence it is crucial to have individuals from different ethnicities represented on CNV maps.


## 2. What is the CNV size that the authors defined? 
The authors defined a size of at least 50 bp that differ in copy number based on the comparison of two or more genomes.

## 3. What are the primary approaches used for CNV detection? And what are the advantages and limitations of these technologies?
Used technologies are mainly Microarrays and NGS. Further Microarrays include CGH and SNP-based arrays. At the moment, no single strategy can capture the entire spectrum of structural variation. One of the limitations is lack of regions which contain segmental duplications, another that duplications are harder to identify in comparison to deletion (SNP-based arrys and NGS approach).

## 4. The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering? 
The cluster filtering was one by using a CNVR-clustering algorithm with a filter choosing only sets od variants where every pair had at least 50% reciprocal overlap. Next it was filtered on the basis of the number of distinct subjects tht carry the variant and the number of distinct studies with at least one variant in the cluster.

The whole filtering process was done to ensure that the final CNVRs excluded false positives as well as study specific artefacts.

## 5. What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively?
- level 1: Supported by only one subject and one study (includes all CNV and CNVRs)
- level 2: Supported by at least two subjects and one study for each variant
- level 12: Supported by at least two subjects and two studies 


## 6. Which percentage of the genome contributes to CNV in inclusive map and stringent map respectively? (Properties of the CNV map)
- inclusive map: 9.5% of the genome involved in losses and gains
- stringent map: totakl genomic variability was 4.8%

Overall, they found CNVRs to be unevenly distributed in the genome and among chromosomes.

## 7. By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their findings in this paper?
By intuition non-coding genes are overall more variable and have a lower impact on the organism, so CNVs are maybe more often found in non-coding regions.

The authors could show this in the paper, especially fitness-altering phenotypes were less variable than the average genome.

## 8. The authors generated a null CNV map and found genes for which at least 85% of the exons were homozygous deleted. What are the functions of these genes? And why did the authors say that they seem to be non-essential? 
They seem to be non-essential because healthy individuals show these losses in the null CNV map. Some of those genes have paralogues in the genome or it affects genes which are not under strong selective pressure (late onset phenotypes). Some examlpes are genes related to age such as genes involved with osteoporosis and multiple sclerosis.

## 9. If you are a medical doctor, how do you use this map as a tool to assess the clinical importance of a CNV? 
As the authors created the CNV map and a list of medically relevant genes from many databases and studies. A medical doctor can either look if a certain CNV of a patient is found in the map and if it overlaps with medically relevant genes. If not found on the map and overlapping with the medically relevant genes, there is a high possibility for clinical relevance.  
