# Genome technologies

## Overview NGS technologies
Since the days of Sanger sequencing, next-generation sequencing technologies have significantly evolved to provide increased data output, efficiencies, and applications. These next generations of technologies can be categorized based on read length.
* Short-read vs. long-read sequencing
  - Short reads are cheaper, can sequence fragmented DNA and show higher sequence fidelity than long reads.
  - With long reads one can sequence also difficult regions such as repeat sequences. It is also very useful for de novo assemblies.

Principle for all NGS technologies: Generating DNA fragments, adding specific adapters to fragments, creating a library, amplification and clustering of fragmented DNA, parallel sequencing of clusters and lastly analysing the genomic data.

## Count based vs. intensity based
* Read count based genotyping
  - Useful to genotype e.g. Transposable Elements because the principle includes sequencing amplicons after PCR and matching the reads to their targeted locations. The matched reads can be counted and higher read numbers can show the presence of a TE or with no reads also the abscence of a TE.

* Intensity based genotyping
  - Makes use of flourescently labeled bases which can be detected during synthesis, enableing sequencing by synthesis.

## Molecular-cytogenetic techniques
* ### Banding analysis
  * G-Banding allows each chromosome to be identified by its characteristic banding pattern. The banding pattern can distinguish chromosomal abnormalities or structural rearrangements, such as translocations, deletions, insertions, and inversions.

* ### SNP, aCGH arrays
  * SNP microarray is the hybridization of fragmented single-stranded DNA to arrays containing hundreds of thousands of unique nucleotide probe sequences.  Each probe is designed to bind to a target DNA subsequence. SNPs are the most frequent type of variation in a genome.
  * Array-based Comparative Genomic Hybridization can locate and identify the gene content of chromosome imbalances. "Normal" and "Looked into" e.g. tumour DNA is differently labelled, competing for hybridisation to their locus of origin on the chromosomes. Loss or gain of genetic material can then be seen.

* ### SKY, M-FISH
  * Spectral karyotyping (SKY) and multiplex FISH (M-FISH) allows visualization of chromosome pairs in a different color at the same time. It helps showing interchromosomal rearrangements.

* ### cCGH
  * Chromosomal Comparative Genomic Hybridization is useful to analyse CNVs relative to ploidy level in the DNA, always compared to a reference gene. Widely used to compare genomes of healthy tissue to the ones of tumour tissue. 

When the techniques are compared to each other, cCGH has an improved resolution over traditional G-banding and FISH. 
