## 1
#!/bin/env/bash
 
#The fastq files are compressed to save space, the following commands will unzip them
gunzip ../data/reads/wildtype/*.fq.gz
gunzip ../data/reads/mutated/*.fq.gz

## 2
#!/bin/env/bash
set -euo pipefail

#Before we can align the reads to the reference with bowtie2, we need to build bowtie2 indices for reference sequence.
bowtie2-build \
#Bowtie is an ultrafast, memory-efficient short read aligner. It aligns short DNA sequences (reads) to the human genome at a rate of over 25 million 35-bp reads per hour. Bowtie indexes the genome with a Burrows-Wheeler index to keep its memory footprint small: typically about 2.2 GB for the human genome (2.9 GB for paired-end).
../data/reference/APC.fa \
../data/reference/APC_idx


## 3
#!/bin/env/bash
set -euo pipefail

#Perform the alignment for the wild type reads against the reference sequence
bowtie2 \
-x ../data/reference/APC_idx \
-1 ../data/reads/wildtype/APC_wt_out1.fq -2 ../data/reads/wildtype/APC_wt_out2.fq \
-S ../data/alignments/wildtype/APC_wt.sam \

#Add read group information to the generated alignment (needed by GangSTR later on)
samtools addreplacerg -r ID:sim20220921 -r PL:wgsim -r SM:wt ../data/alignments/wildtype/APC_wt.sam > tmp.sam
mv tmp.sam ../data/alignments/wildtype/APC_wt.sam

#Perform the alignment for the mutated reads against the reference sequence
bowtie2 \
-x ../data/reference/APC_idx \
-1 ../data/reads/mutated/APC_mut_out1.fq -2 ../data/reads/mutated/APC_mut_out2.fq \
-S ../data/alignments/mutated/APC_mut.sam \

#Add read group information to the generated alignment (needed by GangSTR later on)
samtools addreplacerg -r ID:sim20220921 -r PL:wgsim -r SM:wt ../data/alignments/mutated/APC_mut.sam > tmp.sam
mv tmp.sam ../data/alignments/mutated/APC_mut.sam

## 4
#!/bin/env/bash
set -euo pipefail

#Using samtools view, we convert the alignments generated in the previous step to (binary) BAM format. 
#Then, with samtools sort, we sort the alignments by their starting coordinates.
#Finally, we generate indices for our alignments with samtools index
samtools view -b ../data/alignments/wildtype/APC_wt.sam | samtools sort > ../data/alignments/wildtype/APC_wt.bam
samtools index ../data/alignments/wildtype/APC_wt.bam

samtools view -b ../data/alignments/mutated/APC_mut.sam | samtools sort > ../data/alignments/mutated/APC_mut.bam
samtools index ../data/alignments/mutated/APC_mut.bam


## 5
set -euo pipefail

#We run the GangSTR STR genotyping tool on both our alignments (wildtype and  mutated)
#GangSTR also needs the reference genome and a tab-separated file indicating where STR loci are located
#in the reference genome.
GangSTR --bam ../data/alignments/mutated/APC_mut.bam \
--ref ../data/reference/APC.fa \
--regions ../data/repeats/APC_repeats.tsv \
--out ../results/mutated/mutated

GangSTR --bam ../data/alignments/wildtype/APC_wt.bam 
--ref ../data/reference/APC.fa 
--regions ../data/repeats/APC_repeats.tsv 
--out ../results/wildtype/wildtype

# task 2
samtools

0x1	PAIRED	paired-end (or multiple-segment) sequencing technology    
0x2	PROPER_PAIR	each segment properly aligned according to the aligner     
0x4	UNMAP	segment unmapped     
0x8	MUNMAP	next segment in the template unmapped     
0x10	REVERSE	SEQ is reverse complemented      
0x20	MREVERSE	SEQ of the next segment in the template is reverse complemented     
0x40	READ1	the first segment in the template    
0x80	READ2	the last segment in the template    
0x100	SECONDARY	secondary alignment     
0x200	QCFAIL	not passing quality controls    
0x400	DUP	PCR or optical duplicate    
0x800	SUPPLEMENTARY	supplementary alignment    
use samtools flags  PROPER_PAIR to find correct alignments
