# Genome technologies
![image](https://user-images.githubusercontent.com/114056080/193546710-e399ccfd-8a91-49ce-a6a2-9376ce81d483.png)

## Overview NGS technologies
Next-generation sequencing (NGS) is a massively parallel sequencing technology that offers ultra-high throughput, scalability, and speed. The technology is used to determine the order of nucleotides in entire genomes or targeted regions of DNA or RNA.
> https://www.illumina.com/science/technology/next-generation-sequencing.html#:~:text=What%20is%20NGS%3F,regions%20of%20DNA%20or%20RNA.
## Sanger vs NGS

 Sanger sequencing: \

- Sequencing single genes
- Sequencing 1-100 amplicon targets at the lowest cost
- Sequencing up to 96 samples at a time without barcoding
- Microbial Identification
- Fragment analysis, high throughput genotyping using, for example, SNaPshot
- Microsatellite or STR analysis
- NGS confirmation

NGS:

- Interrogating >100 genes at a time cost effectively
- Finding novel variants by expanding the number of targets sequenced in a single run.
- Sequencing samples that have low input amounts of starting material, using, for example, Ion AmpliSeq library preparation, which requires as little as 10 ng of input DNA
- Sequencing microbial genomes for pathogen subtyping to enable research of critical outbreak situations
## long-read sequencing (LRS)
LRS has the capacity to sequence on average over 10 kb in one single read, thereby requiring less reads to cover the same gene. LRS technologies may offer improvements in the characterization of genetic variation and regions that are difficult to assess with the prevailing NGS approaches. \
>https://www.frontiersin.org/articles/10.3389/fgene.2019.00426/full

Principle for all NGS technologies: Generating DNA fragments, adding specific adapters to fragments, creating a library, amplification and clustering of fragmented DNA, parallel sequencing of clusters and lastly analysing the genomic data.

## short-read sequencing (SRS)
To sequence a large genome like human DNA using NGS, the DNA has to be fragmented and amplified in clones of between 75 base pairs and 400 base pairs.

## Count based vs. intensity based technologies
### count based
 ![image](https://user-images.githubusercontent.com/114056080/193525072-9fe2e545-696c-4c62-ab1e-735b35b26f1f.png)

 

### Intensity based 
  - Intensity based technologies measure fluorescence intensities from dye-labeled molecules to determine the sequence of DNA fragments.

## Molecular-cytogenetic techniques
### Banding analysis
Chromosome banding techniques produce a series of consistent landmarks along the length of metaphase chromosomes that allow for both recognition of individual chromosomes within a genome and identification of specific segments of individual chromosomes.
>https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/0471142905.hg0402s00
### SNP, aCGH arrays
A SNP array is used to detect polymorphisms (SNPs). SNP arrays can detect long contiguous stretches of heterozygosity  \
Comparative genomic hybridization (CGH) is a molecular cytogenetic method for analysing copy number variations (CNVs). This allows for comparison between two sources.
![image](https://user-images.githubusercontent.com/114056080/193545371-08c45e83-00a3-4498-8a30-b6bd727cfd16.png)

### SKY, M-FISH
Multicolor FISH (mFISH), also known as spectral karyotyping (SKY), and multiplex-FISH (M-FISH) permit the simultaneous visualization of each chromosome pair in a different color.
>https://onlinelibrary.wiley.com/doi/abs/10.1002/9781119061199.ch17
### cCGH
  Chromosomal Comparative Genomic Hybridization is useful to analyse CNVs relative to ploidy level in the DNA, always compared to a reference gene. Widely used to compare genomes of healthy tissue to the ones of tumour tissue. 
![image](https://user-images.githubusercontent.com/114056080/193545257-a21c8c62-f5f3-44cf-af91-5f16806998d3.png)
An important advantage of CGH is that it does not require live samples \
FISH provides very high resolution, better than most arrays. \
Of particular interest to the application of SKY and M-FISH is its usefulness in cross-species banding.
