
# T2T-Sequencing and Pangenomes

For previous Full Genome Assembly, short, fragmented methods like shotgun sequencing were used, or like in the GRC project, </br>
bacterial artificial chromosomes (BAC). These methods, while giving, for their time, unparalleled insights into the human genome, <br>
still have their shortcomings.<br>

For example, in the GRCh38 reference assembly, created using BAC, large amounts of unknown sequence remain unaccounted for. <br>
Examples would be repetitive regions like those found in the pericentromeric and subtelomeric parts of the chromosome or structurally <br>
complex parts like segmental duplications, that could not be captured and reconstructed using short reads.<br>
Furthermore, these previous approaches contain certain areas that were incorrectly assembled or computationally <br>
generated due to their patchwork-like approach. <br>

With time, new technologies like long-read shotgun sequencing methods were developed and allowed insights into the <br>
previously challenging segments like centromeric regions. <br>
This sequencing method comes with a high error rate but still allows for reconstruction of previously unmapped regions. <br>
In combination with precise approaches like HiFi sequencing these knowledge gaps could be closed  <br>
and the so-called Telomere-to-Telomere-CHM13 (T2T-CHM13) reference assembly was created, <br>
that contains full documentation of all chromosomal sequences except for the Y chromosome. <br>
This is a major achievement, since this corrects old reconstructive errors and regions like <br>
the human satellite repeat arrays proximate to the centromeres. <br>
With that, new putative coding regions could be predicted yielding information about yet undiscovered <br>
genetic mechanisms and regulatory details, since focusing only on coding regions is not sufficient for throughout scientific investigations. <br>

However, the CHM13 cell line used here could be almost considered haploid since both alleles <br>
are highly congruent and further represents only a single genome which does not adequately account <br>
for genetic diversity among populations. To cover this knowledge gap, efforts to construct a so-called human pangenome are being made. <br>
There, T2T -like sequencing is applied to selected genomes from representative sources to capture global human <br>
genetic diversity. Here, also whole-genome sequencing was used, as well as fully phased genomes containing information <br>
from paternal and maternal haplotypes. <br>
This is a major step into the direction of diverse genomic research.
