# SNP Understanding Question 
### Explore these SNPs 

| number | SNP (rsID) | Gene / Region | Notable Trait / Use |
| ------ | ---------- | ------------- | ------------------- |
| 1      | rs7903146  | _TCF7L2_      | Type 2 Diabetes     |
| 2      | rs1426654  | _SLC24A5_     | Skin Pigmentation   |
| 3      | rs429358   | _APOE_        | Alzheimer's Disease |
| 4      | rs334      | _HBB_         | Sickle Cell Anemia  |
| 5      | rs12913832 | _HERC2_       | Eye Color           |

For each SNP, answer the following:
  
| SNP (rsID)   | Chromosome and position | Variant Type | Reference/ alternate allels| MAF | 
| ------------ | ----------------------- | -------------| -------------------------- | --- | 
| rs7903146    | 10q25.2, 10:112998590 fw |intron variant| C/G/T(forward strand)    |0.40| 
| rs1426654    | Chromosome 15:48134287 fw| missense variant|A/G/T, Ancestral: G    |0.50|
| rs429358     | Chromosome 19:44908684 fw|missense variant |T/C, Ancestral: C      |0.38|
| rs334        | Chromosome 11:5227002 fw | missense variant |T/A/C/G               |0.14|
| rs12913832   | Chromosome 15:28120472 fw| intron variant |  A/C/G, Ancestral: A   |0.43|

* What is the allele frequency of this SNP in different populations (e.g., AFR, EUR, EAS)?
  What differences do you observe? You may include figures (e.g., allele frequency charts from Ensembl) to illustrate your observations.

 |Trait|SNP|Population Frequency|
 |---|---|--------------------|
 |Diabetes T2|rs7903146|<img width="801" alt="grafik" src="https://github.com/user-attachments/assets/91b5a472-b3f8-4453-b18a-d66b57e807af" />|
 |Skin Pigmentation|rs1426654| <img width="839" alt="grafik" src="https://github.com/user-attachments/assets/c9fc7a54-764a-47bc-808d-af6ffa14ffda" />|
 |Alzheimer´s|rs429358|<img width="804" alt="grafik" src="https://github.com/user-attachments/assets/8eee5748-0cf9-4df3-9b22-69beedbc8a5f" />|
 |Sickle Cell Anemia|rs334|<img width="860" alt="grafik" src="https://github.com/user-attachments/assets/17c05635-6214-45d4-8d9f-de88b894775d" />|
 |Eye Color|rs12913832|<img width="832" alt="grafik" src="https://github.com/user-attachments/assets/42662c2d-8acf-46ef-922b-a9234fb15387" />|

  
  
* Choose one GWAS study involving the SNP and describe the associated trait or disease, 
including the reported p-value, effect size (OR or beta), publication details, predicted functional consequences, and include a link to the study from the GWAS Catalog.

[Study Link](https://europepmc.org/article/PMC/10921165) </br>
Ramos-Lopez O. Genotype-based precision nutrition strategies for the prediction and clinical management of type 2 diabetes mellitus. World Journal of Diabetes. 2024 Feb;15(2):142-153. DOI: 10.4239/wjd.v15.i2.142. PMID: 38464367; PMCID: PMC10921165. 

### Background
Globally, type 2 diabetes mellitus (T2DM) is one of the most common metabolic disorders. T2DM physiopathology is influenced by complex interrelationships between genetic, metabolic and lifestyle factors (including diet), which differ between populations and geographic regions. In fact, excessive consumptions of high fat/high sugar foods generally increase the risk of developing T2DM, whereas habitual intakes of plant-based healthy diets usually exert a protective effect. Moreover, genomic studies have allowed the characterization of sequence DNA variants across the human genome, some of which may affect gene expression and protein functions relevant for glucose homeostasis. This comprehensive literature review covers the impact of gene-diet interactions on T2DM susceptibility and disease progression, some of which have demonstrated a value as biomarkers of personal responses to certain nutritional interventions. 

<img width="948" alt="grafik" src="https://github.com/user-attachments/assets/e01e19ae-a8a2-45b6-b67f-b190d6c15efa" />

### From: GENETIC POLYMORPHISMS AS BIOMARKERS OF GLYCEMIC RESPONSES TO DIETARY ADVICE

Also, obese individuals who were homozygous for the T-risk allele of the transcription factor 7 like 2 (TCF7L2) rs7903146 polymorphism and consumed a high-fat (40%-45% E) diet, underwent smaller reductions in HOMA-estimated insulin resistance (HOMA-IR)[^44].

### From cited study in review:
Transcription factor 7-like 2 (TCF7L2) rs7903146 associates with type 2 diabetes and may operate via impaired glucagon-like peptide 1 secretion, which is stimulated more by fat than by carbohydrate ingestion.

### Design: 
European, obese participants (n = 771) were randomly assigned to receive an HF or an LF diet. Body weight, fat mass (FM), fat-free mass (FFM), waist circumference (WC), resting energy expenditure (REE), fasting fat oxidation in percentage of REE (FatOx), homeostasis model assessed insulin release (HOMA-β), and HOMA–insulin resistance (HOMA-IR) were determined at baseline and after the intervention; 739 individuals were genotyped for rs7903146.
### Results: 
Average weight loss was 6.9 kg with the LF and 6.6 kg with the HF (difference between diets, NS) diet.
Among individuals who were **homozygous for the T-risk allele**, those in the **HF diet group** experienced **smaller weight losses** (Δweight) (2.6 kg; **P = 0.009; n = 622**), **smaller ΔFFM (1.6 kg; P = 0.027; n = 609)**, **smaller ΔWC (3.3 cm; P = 0.010; n = 608)**, and a **smaller ΔHOMA-IR (1.3 units; P = 0.004; n = 615)** than did the LF diet group. 
For C allele carriers, there were no differences between the HF and LF diet groups. For the HF diet group, each additional T allele was associated with a reduced loss of FM (0.67 kg; P = 0.019; n = 609). 
**TCF7L2 rs7903146 was not associated with ΔREE, ΔFatOx, ΔHOMA-β, or dropout.**

[^44]: https://www.sciencedirect.com/science/article/pii/S0002916523016428?via%3Dihub
