
 
## Why is important to construct a CNV map on health individuals of various ethnicities?
Essentially, pathogenic CNVs continue to be described in differ- ent classes of disease15. Clinical geneticists need to discriminate pathogenic or high-risk variants from benign variants in patients, and they do so with information about CNVs found in apparently healthy individuals.

## What is the CNV size that the authors defined? (Box 1 mentioned in introduction)
Microscopic and submicroscopic variants from 50 bp to 3 Mb.

*(Noted: The CNV size definition is still under debate and may be different in other literature)*

## What are the primary approaches used for CNV detection? 
Microarrays and next-generation sequencing (NGS) are now the primary approaches used for CNV detection.

## And what are the advantages and limitations of these technologies? (CNV discoveries)
Currently, no single discovery strategy can capture the entire spectrum of structural variations in the genome:

- Numerouus platforms have a lack of probes in regions of the genome that contain segmental duplications — which predispose to structural variation — and are thus unable to recognize many CNVs that have been implicated in human disease.

- Deletions are generally easier to identify, and methods vary in their ability to detect deletions or duplications; for example, more duplications are missed by SNP-based array platforms and NGS approaches than by array CGH.

- The size distribution of variants called by different platforms varies. Compared with array-based detection, studies that use sequencing-based methods, particularly NGS, call smaller variants and are biased towards the detection of deletions. Sequencing-based approaches have both high sensitivity and the ability to provide accurate sequence-level breakpoint resolution, whereas array-based approaches have a limited resolution capacity.

## The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). What are the two criteria for cluster filtering? 
The number of distinct subjects that carry the variant and the number of distinct studies with at least one variant in the cluster 

## And why did they do this filtering? (The CNV map)
To combine the high-confident variants from different studies into merged CNVRs

## What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively? (The CNV map)
stringency level 1: at least one subject and one study for each variant
stringency level 2: at least two subjects and one study for each variant
stringency level 12: at least two subjects and two studies for each variant

## Which percentage of the genome contributes to CNV in inclusive map and stringent map respectively? (Properties of the CNV map)
|map | contribution
|-|-|
|inclusive map | 9.5%|
|stringent map | 4.8%|

## By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their findings in this paper? (Functional impact of CNV)
My intuition would tell me that non-coding genes would be more variable (no selection pressure).

The exons of all genes (as defined by RefSeq62) were more variable than the genome average (that 50
is, the background). The exons of non-coding genes had the highest proportion of copy number variable sequence, which is higher than the exons of protein-coding genes. By contrast, exons of many of the constrained gene sets — particularly those associated with diseases (OMIM, ISCA and cancer) or other fitness-altering phenotypes — were less variable than the genome average.

Exons in dominant genes showed no significant differences in variability compared with those in recessive genes

Promoters were enriched in CNVRs compared with the entire genome

Enhancers were impoverished in CNVRs. PhastCons elements were found in relatively few CNVRs, as were ultra-conserved elements, which are under strong selection against mutation. CpG islands were more variable than the genome background.

## The authors generated a null CNV map and found genes for which at least 85% of the exons were homozygous deleted. What are the functions of these genes? And why did the authors say that they seem to be non-essential? (Homozygous deleted genes)
These genes would seem to be non-essential, as they could be missing from the genomes of apparently healthy individuals.
The function of non-essential genes may be redundant (14 have paralogues), or they may be related to late-onset phenotypic consequences that do not substantially reduce the fitness. Some of the non-essential genes from our list are described in OMIM as age-related phenotypes (UDP glucuronosyltransferase 2 family polypeptide B17 (UGT2B17) for osteoporosis75), Rh-negative blood type (RHD76), delayed AIDS from HIV-1 infection (KIR3DL1 (killer cell immunoglobulin-like receptor, three domains, long cytoplasmic tail, 1)77), susceptibility to multiple sclerosis (the major histocompatibility complex gene HLA-DRB1 (REF. 78)), autoimmune dis- ease (HLA-DQA1 (REF. 79) for coeliac disease), and pla- cental function and fetal wellbeing (pregnancy-specific beta-1-glycoprotein 1 (PSG1)80).

## If you are a medical doctor, how do you use this map as a tool to assess the clinical importance of a CNV? (Clinical application of the CNV map part in Discussion)
This map can be used as a tool in the investigation of CNVs for medical applications, for example to assess the clinical importance of a CNV found in a case subject. The criteria could include:
(i) whether it is found among the CNVRs of the CNV 2
map and (ii) whether it overlaps with medically relevant genes

