
 
## Why is important to construct a CNV map on health individuals of various ethnicities?
(Introduction)
## What is the CNV size that the authors defined? (Box 1 mentioned in introduction)
microscopic and submicroscopic variants from 50 bp to 3 Mb

Noted: The CNV size definition is still under debate and may be different in other literature

## What are the primary approaches used for CNV detection? And what are the advantages and limitations of these technologies? (CNV discoveries)
Microarrays and next-generation sequencing (NGS) are now the primary approaches used for CNV detection.

Currently, no single discovery strategy can cap- ture the entire spectrum of structural variations in the genome

Numerouus platforms have a lack of probes in regions of the genome that contain segmental duplications — which predispose to structural variation — and are thus unable to recognize many CNVs that have been implicated in human disease

Deletions are generally easier to identify, and methods vary in their ability to detect deletions or duplications; for example, more duplications are missed by SNP-based array platforms and NGS approaches16,40 than by array CGH.

The size distribution of variants called by differ- ent platforms varies. Compared with array-based detection, studies that use sequencing-based methods, particularly NGS, call smaller variants and are biased towards the detection of deletions16,43. Sequencing-based approaches have both high sensitivity and the ability to provide accurate sequence-level breakpoint resolution, whereas array-based approaches have a limited resolu- tion capacity.

## The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering? (The CNV map)
the number of distinct subjects that carry the variant and the num- ber of distinct studies with at least one variant in the cluster

to combine the high-confident variants from different studies into merged CNVRs

## What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively? (The CNV map)
at least one subject and one study (map with the stringency level 1)
at least two subjects and one study for each variant (strin- gency level 2)
at least two subjects and two studies (stringency level 12)

## Which percentage of the genome contributes to CNV in inclusive map and stringent map respectively? (Properties of the CNV map)
|map | contribution
|-|-|
|inclusive map | 9.5%|
|stringent map | 4.8%|

## By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their findings in this paper? (Functional impact of CNV)

## The authors generated a null CNV map and found genes for which at least 85% of the exons were homozygous deleted. What are the functions of these genes? And why did the authors say that they seem to be non-essential? (Homozygous deleted genes)


## If you are a medical doctor, how do you use this map as a tool to assess the clinical importance of a CNV? (Clinical application of the CNV map part in Discussion)


