
# Task: Exploring Genome Resources


primary deposition databases
interpreted databases (e.g. variant annotations...)
suggestion: VICC paper (Wagner et al.)
Wagner et al (2020): A harmonized meta-knowledgebase of clinical interpretations of somatic genomic variants in cancer

## make some notes about different genome resources and their primary use
### Reference Genome Resources
* genome.ucsc.edu UCSC genome browser
  * open source
  * human and closely related species, as well as mice
  * has its own alignment tool (BLAT), PCR design and genome annotation tools
  * multi level comparison, gene control/expression, SNPs, locus and mutations, conservation
  * full sequenced sars cov genome
  * originated from the human genome project
  * widely used and referenced
* HUMAN GENOME RESOURCES AT NCBI
  * international dna data bank
  * massive size of any type of genetic, protein and compound available, mutations and genomic differences and consequences
  * only human genome
  * vast array of searching and sorting tools (BLAST etc.)
  * contains dbVar, clinVar, dbSNP as subsection for respective content
  * virtual library with literature and reference material
  * difficult to navigate due to size
* Ensemble
  * evolutionary library
  * focus and phylogenetics, change and conservartion
  * human and vertabrates, even some invertabrates
  * genome, variation and consevation

### Genome Variant Data resources
* NCBI:dbSNP
  * subdirectory of the NCBI for SNPs
* NCBI:dbVAR
  * subdirectory of the NCBI for Variable sequences
* NCBI:ClinVar
  * subdirectory of the NCBI for genome changes and approximate consequences
* EMBL-EBI:EVA
  * variation browser with prediction and data pertaining to any human genome
  * multitude of animals, vertebrates and mammals
  * directly linked SNPs to relevant papers 
* Ensembl
  * portal for many things genomic
### Cancer genomics resources
* cosmic
  * strong citations in cancer research 
  * mutation predictions, gene fusion tools, etc
* ClinGen
  * platform for exchange and sorting of cancer research
  * sorting mutations to effect and a more limited prediction
  * state funded
* ClinVar
  * basis for variant <-> disease associations
* CGAP
  * annotiations of cancer mutation effects
  * privately owned but open
  * human cancer vatiations and any other genetic material or substance pertaining to it

### Cancer Variant resources
* arrayMap/Progenetix'
  * meta analysis og human cancers
  * smaller size doue to lack of beins to jit
