
# 01_unzip_reads.sh

This script unzips the fastq files in data/reads.
Tools used:
  - gunzip

File formats:
  - fastq
  - GZ

# 02_bowtie2_index.sh

Build bowtie2 indices for reference sequence.
Tools used:
  - bowtie 2

File formats:
  - fasta
 
# 03_alignment_bowtie2.sh

Perform the alignment for the wild type and mutated reads against the reference sequence
Add read group information to the generated alignment (needed by GangSTR later on)
Tools used:
  - bowtie 2
  - samtools

File formats: 
  - fastq
  - sam

# 04_process_alignment.sh

Using samtools view, we convert the alignments generated in the previous step to (binary) BAM format. 
Then, with samtools sort, we sort the alignments by their starting coordinates.
Finally, we generate indices for our alignments with samtools index
Tools used:
  - samtools view, sort & index

File formats:
  - sam
  - bam

# 05_run_gangstr.sh 

We run the GangSTR STR genotyping tool on both our alignments (wildtype and  mutated)
GangSTR also needs the reference genome and a tab-separated file indicating where STR loci are located
in the reference genome.

Tools used:
  - GangSTR

File formats:
  - fasta
  - bam
  - tsv
