# Answers Human Pangenome Reference 
[A draft human pangenome reference](https://doi.org/10.1038/s41586-023-05896-x) 

## Why ist the Pangenome project interesting?
The human pangenome project is intereting because it is a massive progress to the original human genome reference from 2003. It includes 47 diploid genomes from people from different geographical and ethnical origin to refelct the genomic diversity of the human species (The goal is to reach 700  haplotypes from 350 individuals). The pangenomes completed the 8% missing DNA parts of the original, has a less error is detecting small variations and detects more structure variation 

"The pangenome project aims to provide accessible high-quality genomes that represent the diversity of the human population." Source: [Human pangenome project](https://humanpangenomeproject.org)


## What makes the t2t approach different to previous approaches? What is being sequenced and why?
In 2022 the T2T project published the first human genome (Telomere to telomere) without gaps (except rDNA). They used PacBio HiFI and Oxford Nanopore long-read sequencing, Bionan optical maps and Hi-C.
The t2t approach made it possible to completly sequencing the human genome and find more SNPs and CNVs, that were missing in the original human reference genome. 


## What are the future benefits of the pangenome for genome data analysis?
"The new human pangenome reference will help researchers analyze human genome sequences and understand how genomic variants influence human health. Using the human pangenome reference, researchers are already finding genomic variants that were not previously identified using the older human genome reference sequences." 
Source: [Why do we need a new human pangenome reference? ](https://www.genome.gov/about-genomics/educational-resources/infographics/pangenome#:~:text=The%20new%20human%20pangenome%20reference%20will%20help%20researchers%20analyze%20human,older%20human%20genome%20reference%20sequences)

A previous problem was the missing of e.g. rare or populationsspecific allels, because of the lack of variation in the reference genome. Mapping of sequences, especially in regiosn in which sequencing was harder (e.g. repetitive sequences, centromeres), becomes more accurate. 



