# Why is important to construct a CNV map on health individuals of various ethnicities?
- will aid the interpretation of new CNV findings for both clinical and research applications
- continuous spectrum of phenotypic effects of CNV, from adaptive traits to embryonic lethality
# What is the CNV size that the authors defined?
- he current map includes microscopic and submicroscopic variants from 50 bp to 3 Mb
# What are the primary approaches used for CNV detection? And what are the advantages and limitations of these technologies?
- microarray and NGS
- array-based detection methods are suitable for studying quantitative variants
- NGS based methods call smaller variants and are biased towards the detection of deletions
- sequencing-based approaches have both high sensitivity and the ability to provide accurate sequence-level breakpoint resolution
- array-based approaches have a limited resolution capacity
# The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering?
- sets of variants in which every possible variant pair had at least 50% reciprocal overlap
- number of distinct subjects that carry the variant and the number of distinct studies with at least one variant in the cluster
- the filter based on the number of subjects excluded singletons, and variants that are supported by a larger number of subjects are less likely to be false positives
- the filter based on the number of studies ensured the exclusion of potential study-specific artefacts
# What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively?
- stringency level 1: at least one subject and one study includes all CNV and CNVRs
- stringency level 2: at least two subjects and one study for each variant 
- stringency level 12: at least two subjects and two studies
# Which percentage of the genome contributes to CNV in inclusive map and stringent map respectively?
- inclusive map: 9.5 %
- stringent map: 4.8 %
# By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their findings in this paper?
- intuition: non-coding genes, the direct impact is not that high as in protein coding genes
- paper:  
  - exons of all genes were more variable than the genome average
  - exons of non-coding genes had the highest proportion of copy number variable sequence
# The authors generated a null CNV map and found genes for which at least 85% of the exons were homozygous deleted. What are the functions of these genes? And why did the authors say that they seem to be non-essential?
- non essential: they could be missing from the genomes of apparently healthy individuals
- function: redundant, may be related to late-onset phenotypic consequences that do not substantially reduce the fitness
