# Task: Exploring Genome Resources

### primary deposition databases
Primary databases only contain information for sequence or structure.[^1].
  [^1]: https://www.enago.com/academy/biological-databases-an-overview-and-future-perspectives/

### interpreted databases (e.g. variant annotations...)
Contain data derived from primary databases. They store informasion like for example conserved sequences, active site residues and more.
Example: ANNOVAR, which offers 3 types of annotations (gene-based annotation, region-based annotation and filter-based annotation)[^2].
  [^2]: https://www.nature.com/articles/nprot.2015.105
  
### VICC paper (Wagner et al.)
Wagner et al (2020): A harmonized meta-knowledgebase of clinical interpretations of somatic genomic variants in cancer make some notes about different genome resources and their primary use[^3].
  [^3]: https://www.nature.com/articles/s41588-020-0603-8
Abstract: 
'Precision oncology relies on accurate discovery and interpretation of genomic variants, enabling individualized diagnosis, prognosis and therapy selection. 
We found that six prominent somatic cancer variant knowledgebases were highly disparate in content, structure and supporting primary literature, impeding consensus when evaluating variants and their relevance in a clinical setting. 
We developed a framework for harmonizing variant interpretations to produce a meta-knowledgebase of 12,856 aggregate interpretations. 
We demonstrated large gains in overlap between resources across variants, diseases and drugs as a result of this harmonization. 
We subsequently demonstrated improved matching between a patient cohort and harmonized interpretations of potential clinical significance, observing an increase from an average of 33% per individual knowledgebase to 57% in aggregate. 
Our analyses illuminate the need for open, interoperable sharing of variant interpretation data. 
We also provide a freely available web interface (search.cancervariants.org) for exploring the harmonized interpretations from these six knowledgebases.'
