# Primary deposition databases 

## [Breast Cancer Gene-Expression Miner v5.0](http://bcgenex.ico.unicancer.fr/BC-GEM/GEM-Accueil.php?js=1)
### Primary use
It is a tool to explore annotated breast cancer transcriptomic data (DNA microarrays and RNA-seq) of published data.
It can be used to explore the expression of genes of interest in breast cancer. 
The information that is available comprises characteristics, functions and role of genes that play a role in oncogenesis. 

## [MSK's Precision Oncology Knowledge Base](https://www.oncokb.org/)
### Primary use
OncoKB is a knowledge base of somatic molecular alterations in tumors. The information stored in the KB can be used
for predicting the drug response for the FDA, for making guidelines of the National Comprehensive Cancer Network (NCCN) and recommendations for specific diseases. 



# Interpreted databases
## [Cancer Genome Interpreter](https://www.cancergenomeinterpreter.org/home)
[Reference](https://link.springer.com/article/10.1186/s13073-018-0531-8)
### Functions
* Inteprets cancer genomes that have recently been sequenced
* Annotates potential tumor variants that could act as drivers
* Describes possible effects on treatment response

### Primary Use
Useful to find recent tumor genome sequences and their annotations, hence the role they possibly play in tumorigenesis and treatment (tumor suppressors and oncogenes). It provides a systematic approach to the interpretation of cancer genomes to standardize the different resources available in the field.
* Identification of known tumorigenic genomic alterations (i.e. point mutations, indels, CNA and gene fusions)
* Assessment of variants that have an unclear significance
* Annotation of tumor variants that are biomarkers of drug response.



# VICC paper (search.cancervariants.org)
Variant Interpretation for Cancer Consortium: It is a web interface that combines and interprets six different
knowledgebases. This is necessary because each knowledgebase is different in terms of content, structure and 
literature. This leads to problems when trying to make inferences between patient cohorts and clinical
significance. Hence, there is a need for sharing of data and making the different databases compatible. 
They combine data from following knowledgebases:
* Cancer Genome Interpreter Cancer Biomarkers Database (CGI)
* Clinical Interpretation of Variants in Cancer (CIViC)
* Jackson Laboratory Clinical Knowledgebase (JAX-CKB)
* MolecularMatch (MMatch)
* OncoKB
* Precision Medicine Knowledgebase (PMKB)

The standard and guidelines are based on the recommendations by AMP/ASCO/CAP.
This collection of information can be used by experts that match therapy to the specific tumor type, for regulatory 
processes of FDA genomic tests and for the identification of biomarkers of different diseases that deliminate future 
areas of study. 
