# Task: Reading up on genome technologies

## Next Generation Sequencing (NGS) technologies:
-	Unlike traditional Sanger sequencing, NGS allows for massively parallel sequencing of millions of DNA fragments simultaneously.
-	Limitations: interpretation/analysis and storage!
-	It enables rapid and cost-effective sequencing of DNA and RNA
-	Some examples for NGS technologies: Illumina sequencing, ion torrent sequencing, PacBio sequencing, Nanopore sequencing

## count based vs. intensity based methods
-	**count-based**: counting the number of times a particular DNA sequence appears in a sample (counting the reads). Count-based methods are prevalent in NGS technologies like Illumina sequencing where the number of reads (DNA fragments) are counted
-	**intensity-based**: measuring the intensity of signals generated by labelled nucleotides ïƒ  the intensity of fluorescent signals is proportional to the quantity of molecules present in a sample.

## Long read technologies:
-	can sequence repetitive elements and structural variations which are difficult to do for short read technologies
-	Long reads facilitate the assembly of complete genomes
-	Examples for long-read technologies: PacBio and Oxford Nanopore

## Short read technologies:
-	fast and cost-efficient
-	high base accuracy
-	good for quantifying gene expression levels and detecting SNPs
-	Example for short-read technologies: Illumina sequencing

## (molecular)-cytogenetic techniques: 
-	**banding analysis**: banding means staining chromosomes to create a characteristic pattern of light and dark bands. This banding pattern allows for the identification of individual chromosomes and the detection of structural abnormalities. Used historically but still applied as well in current clinics.
-	**SNP arrays**: SNP arrays detect variations in single nucleotides and are used to identify CNVs.
-	**aCGH arrays**: aCGH compares the genomic DNA of a test sample against a reference sample to identify chromosomal gains or losses.
-	**SKY**: involves hybridizing fluorescently labeled chromosome-specific probes to metaphase chromosomes.
-	**M-FISH**: uses multiple differently colored fluorescent probes to label specific chromosome regions, enabling the detection of complex chromosomal rearrangements.
-	**chromosomal CGH**: compares the DNA copy number between a test sample and a reference sample, identifying chromosomal imbalances.

## "T2T genome" 
-	Refers to Telomere-to-Telomere (T2T) Consortium's initiative to produce a high-quality and complete representation of the human genome.
-	Requires a combination of advanced technologies like high-quality und fidelity short read sequencing as well as long-read sequencing to be able to obtain a gapless sequencing of the entire human genome
