---
title: "bio392-cnv-freq"
author: "Leila Osterwalder"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
#install.packages("devtools")
devtools::install_github("progenetix/pgxRpi")
```

# Tumor information

## Name and NCIt code
invasive breast carcinoma:
The NCIt code is C9245

## Definition
A carcinoma that infiltrates the breast parenchyma. The vast majority are adenocarcinomas arising from the terminal ductal lobular unit. Often, the invasive adenocarcinoma co-exists with ductal or lobular carcinoma in situ. It is the most common carcinoma affecting women. Invasive breast cancer can spread to other parts of the body through the blood and lymph systems.

# Access the CNV frequency data

## load library
```{r}
library(pgxRpi)
```

## access data
```{r}
# NCI Thesaurus, search for cancer name
freq <- pgxLoader(type='frequency', output='pgxfreq', filters="NCIT:C9245")
```

The metadata information is
```{r}
freq$meta
```

The actual data looks like
```{r}
freq$data$`NCIT:C9245`[c(1,3),]
```

# Visualisation

```{r}
pgxFreqplot(freq)
# you can see which chromosomes have frequent gain or loss
# blue is deletion frequency, yellow the duplication frequency
# dashed line represents the centromer, links von Linie: p Arm, rechts: q Arm
```

```{r}
# specify parameter (what chromosome do I wanna have a closer look at)
pgxFreqplot(freq, chrom=8)
```

```{r}
# query multiple chromosomes
pgxFreqplot(freq, chrom=c(4,9,15))
```

# Analysis

According the plot, there are very frequent duplication on chromosomes 1q and 8q (over 50%). Rather frequent (cut-off at 25% frequency) we find on chromosomes 16p and 20q. Frequent deletions (threshold ca. 25%) are found on chromosomes 4, 6q, 8p, 9p, 9q, 11q, 13q, 16q, 17p, 18q, 22q, Xp and Xq.

One [study](https://www.nature.com/articles/bjc1998583) detected frequent gains on chromosomal regions 1q, 8q, 16p, 19, 20q and X as well as frequent losses on 1p, 4q, 5q, 6q, 9p, 11q, 13q and 16q. 

There are overlaps (between progenetix and literature e.g. nature.com): which chromosomes were the same
Surprisingly the study found frequent gains on the X chromosome, whereas the data from Progenetix suggests clearly frequent losses on the X chromosome.

The differences in results between the study and the Progenetix database could be due to several reasons. First, the study from nature.com was published in 1998 so the chromosomal regions prevalently affected by mutations might have shifted a bit over time. Moreover the method used to detect chromosomal abnormalities in the nature study was comparative genomic hybridization. Detecting DNA gains and losses in the Progenetix database may involve a combination of techniques, including array-based comparative genomic hybridization (aCGH), next-generation sequencing (NGS), and other molecular genetic methods. As a last possible explanation for the differences one couldlist the number of samples used to get the data. In the study there were only 39 carcinomas used, whereas in Progenetix, the data from several thousand samples come together.

