#!/usr/bin/env bash
set -euo pipefail

# Many tools that work with reference genome sequences require an
# index to allow random access. In our case: GangSTR expects the
# reference genome to have an index generated by SAMtools, which
# is what we generate here.
samtools faidx ../data/reference/APC.fa

# BWA-MEM2, the aligner that we will use to map the sequencing reads 
# to the reference genome, also needs an index.
bwa-mem2 index ../data/reference/APC.fa
