### 1. Why is it important to construct a CNV map on healthy individuals of various ethnicities?

it will aid the interpretation of new CNV findings for both clinical and research applications.

### 2. What is the CNV size that the authors defined?

larger than 50 bp. smaller elements are knows as indels

### 3. What are the primary approaches used for CNV detection? And what are the advantages and limitation of these technologies?

Microarrays and NGS are the primary approaches used for CNV detection. \
No single strategy can capture the entire spectrum of structural variations in the genome. \
The arraybased detection methods are suitable for studying quantitative variants. But many small variants are missed.\
NGS methods call smaller variants and are biased towards detection of deletions

### 4. What are the two criteria for cluster filtering? And why did they do this filtering?

Used an algorithm to identify sets of variants in which every possible variant pair had at least 50% reciprocal overlap|
Clusters were then filtered on the basis of the number of distinct subjects that carry the variant and the number of\
distinct studies with at least on variant in the cluster. \
It was filtered because it ensured that structurally different CNV's were not merged

### 5. What are the thresholds in stringency lvl1, lvl2 and lvl12 respectively?

stringency lvl1: at least one subject and one study included all CNV's and CNVR's\
stringency lvl2: at least two subjects and one study for each variant
stringency lvl12: at least two subjects and two studies

### 6. Which percentage of the genome contributed to CNV in inclusive map and stringent map respectively?

stringent map: 4.8%
inclusive map: 9.5%

### 7. By your intuition, which kind of genes are more variable between protin-coding genes and non-coding genes? How about the paper's findings?

Since variability in coding genes will lead to defective proteins i assume they would be selected against in evolution. non-coding genes may not have such a big impact when they are defective.
The exons of non-coding genes had the highest proportion of CNV's. Which is what i thought

### 8. What are the functions of the genes of the null CNV map? why did the authors say that they seem to be non-essential?

because they could be missing from the genomes of apparently healthy individuals\
theyir function may be redundant or they may be related to late-onset phenotypic consequences that do not substantially reduce the individuals fitness.\
some of them are described in OMIM as age-related phenotypes for osteoporosis, Rh- bloodtype, delayed AIDS from HIV-1 infection, susceptibility to multiple sclerosis, autoimmune disease \
and placental function and fetal wellbeing\

### 9. If you are a medical doctor, how do you use this map as a tool to asses the clinical importance of a CNV?

one could check whether the CNV is found among the CNVR's of the CNV map and whether it overlaps with medically relevant genes.\
the researchers compiled a list of 935 medically relevant genes that have been shown to involve CNV's. \
A CNV found in a patient that overlaps with any of these genes while note being found in the CNV map would suggest medical importance.









