
Paper link: https://www.nature.com/articles/nrg3871
 
1. Why is important to construct a CNV map on health individuals of various ethnicities?

2. What is the CNV size that the authors defined? (Box 1 mentioned in introduction)
  Noted: The CNV size definition is still under debate and may be different in other literature
3. What are the primary approaches used for CNV detection? And what are the advantages and limitations of these technologies? 
  (CNV discoveries)
4. The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). 
  What are the two criteria for cluster filtering? And why did they do this filtering? (The CNV map)
5. What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively? (The CNV map)
6. Which percentage of the genome contributes to CNV in inclusive map and stringent map respectively? (Properties of the CNV map)
7. By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their 
  findings in this paper? (Functional impact of CNV)
8. The authors generated a null CNV map and found genes for which at least 85% of theexons were homozygous deleted. 
  What are the functions of these genes? And why did the authors say that they seem to be non-essential? (Homozygous deleted genes)
9. If you are a medical doctor, how do you use this map as a tool to assess the clinicalimportance of a CNV? (Clinical application of the CNV map part in Discussion)

copy number variations involve unbalanced rearrangements which increase or decrease the DNA content \
CNVs size is defined larger than 50bps

a trait can be adaptive or maladaptive in different environmental contexts \
maladaptive CNVs have been associated with conditions such as autism, schizophrenia, Crohn's disease, rheumatoid arthritis, type 1 diabetes, obesity and numerous developmental diseases 

high-resolution CNV maps have been constructed to study the impact of CNVs on complex human traits, including diseases \
aim of this Analysis is to create a CNV map of the human genome for variations that are not associated with adverse phenotypes 

microarrays and next-generation sequencing are the primary approaches used for CNV detection \
with the advent of NGS technologies, new tools can detect CNVs from the data \
array-based detection methods are suitable for studying quantitative variants \
deletions are easier to identify \
sequencing-based approaches have high sensitivity and the ability to provide accurate sequence-level breakpoint resolution, whereas array-based approaches have a limited resolution capacity 

**The CNV map** \
aim of the CNV map is to document the variability of the human genome in healthy individuals from various populations \
sanger sequencing can detect CNV breakpoints with single-nucleotide resolution \
CNVs from diffrent studies are used \
clusters were then filtered on the basis of the number of distinct subjects that carry the variant \
additional information about the ethnic origin, number of studies and number of variants are added in each CNVR 

**Properties of the CNV map** \
the inclusive CNV map of the human genome comprises 3,132 CNVR gains and 23,438 CNVR losses \
3.7% of the human genome is copy number variable -> slightly lower than estimated (4.8%) \
difference because only 40 samples of European and African ancestry were analysed \
higher variability in inclusive map could be explained by the fact that more samples from a wider range of ethnic backgrounds were included \
total genomic variability for the map is 4.8% (3.6% losses, 2.3% gains), 50% less than that of the inclusive map \
most CNVs 300-3000bp, losses are a little bit maller than gains \
CNVRs are unevenly distributed in the genome and on the chromosomes \
chr 22 and the Y chr showed the highest proportion of variability for gains \
for losses highest proportion is on chr 19, 22 and Y \
chr 3 and 18 showed the lowest proportion of variability in this map \
Y chr seems to be copy number stable, it is difficult to design probes for it 

**Functional impact of CNV** \
different genomic elements are expected to be under different degrees of constraint for variation in copy number \
losses contributed more than gains to the overall variation \
exones are more variable than the genome average \
exons of non-coding genes had the highest proportion of copy number variable sequence \
exons associated with diseases or other fitness altering phenotypes are less variable \
cancer related genes are under strong negative selection \








