{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "35f70351",
   "metadata": {},
   "source": [
    "# Genotyping short tandem repeat (STRs) in sequencing data with GangSTR\n",
    "### BIO392 30.09.2022\n",
    "Contact: Max Verbiest (maxadriaan.verbiest@uzh.ch)\n",
    "\n",
    "In this notebook, we will do a hands-on introduction to genotyping STRs in alignments using GangSTR. We will subsequently use GangSTRs output to look for a mutation in on of the two samples we analyse.\n",
    "\n",
    "The situation is as follows:\n",
    "\n",
    "You have two alignments resulting from the sequencing of two samples from a colorectal cancer patient. You know that one sample was taken from healthy tissue, and one from the patients tumour. The only problem is... you forgot which sample came from which tissue! (very embarrassing)\n",
    "Your task is now to figure out which of the two samples is most likely to come from the tumour, based on the STR genotypes that you will determine with GangSTR."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "537509fc",
   "metadata": {},
   "source": [
    "### 1: Run GangSTR\n",
    "\n",
    "First of all, we need to run GangSTR. Open up your terminal and follow the steps from the image below:\n",
    "\n",
    "\n",
    "![title](img/run_gangstr.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e26d4d42",
   "metadata": {},
   "source": [
    "If all went well, GangSTR should have generated output files in the 'results' directory. Let's check:\n",
    "\n",
    "\n",
    "![title](img/check_output.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6db82b89",
   "metadata": {},
   "source": [
    "Feel free to take a look at the contents of these output files. The most relevant files for us are the ones with '.vcf' extension (Variant Call Format). For now, just know that these are output files generated by GangSTR that will tell us the STR genotypes at the loci we specified. In a later session we will take a closter look at file formats related with the analysis of biological sequences."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be8ab8e1",
   "metadata": {},
   "source": [
    "### 2: Load GangSTR output as a pandas DataFrame"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab6a8f00",
   "metadata": {},
   "source": [
    "First, we will load the required libraries and scripts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "32863594",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "sns.set_context(\"poster\")\n",
    "\n",
    "import gangstr_utils\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b8423e8",
   "metadata": {},
   "source": [
    "Next, we load the GangSTR vcf files into Pandas DataFrames.\n",
    "\n",
    "As a sidenote: for this notebook I wrote a *very* rudimentary function to parse these vcf files and combine them with our previously generated set of reference STR loci. This is to reduce the number of dependencies needed to be installed for this notebook to work. In any real-life scenario, it is much wiser to use some third-party vcf parsing library, such as [cyvcf2](https://brentp.github.io/cyvcf2/)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9cf8a114",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_str_loci = pd.read_csv(\n",
    "    \"../data/repeats/APC_repeats.tsv\", \n",
    "    sep=\"\\t\",\n",
    "    header=None,\n",
    "    names=[\"chr\", \"start\", \"end\", \"unit_len\", \"unit\"]\n",
    ")\n",
    "\n",
    "df_gangstr_results_s1 = gangstr_utils.load_gangstr_output(df_str_loci, \"../results/sample_1/sample_1.vcf\")\n",
    "df_gangstr_results_s2 = gangstr_utils.load_gangstr_output(df_str_loci, \"../results/sample_2/sample_2.vcf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec6c87f5",
   "metadata": {},
   "source": [
    "Let's print the first 10 entries of one of these dataframes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "49329a83",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_gangstr_results_s1.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1823f724",
   "metadata": {},
   "source": [
    "This is what every column in this dataframe represents:\n",
    " - 'chr': the chromosome the STR is located on\n",
    " - 'start': the first position of the STR \n",
    " - 'end': the last position of the STR\n",
    " - 'unit_len': the length of the repeating DNA motif\n",
    " - 'unit': the sequence of the repeating DNA motif \n",
    " - 'ref': the copy number that this STR locus has in the reference genome\n",
    " - 'alt': the copy number that GangSTR found for this STR locus in the alignment\n",
    " \n",
    "Using this information, we can start our hunt for mutations!"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72d0cc82",
   "metadata": {},
   "source": [
    "### 3: Combine information from both samples and look for mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b814cf0d",
   "metadata": {},
   "source": [
    "We will first combine data from our two samples into one DataFrame.\n",
    "\n",
    "**Bonus exercise:**\n",
    "If you are already familiar with Pandas, or would like to learn, you can try to merge the two dataframes yourself! The [pandas documentation pages](https://pandas.pydata.org/docs/) have a ton of information and tutorials. Alternatively, you can uncomment the first few lines in the block below, to use a pre-written merging function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d0052b77",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_comparison = gangstr_utils.merge_samples(\n",
    "#     sample1=df_gangstr_results_s1,\n",
    "#     sample2=df_gangstr_results_s2\n",
    "# )\n",
    "\n",
    "\n",
    "df_comparison.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d447b544",
   "metadata": {},
   "source": [
    "Now that we have both samples in the same DataFrame, we can easily determine if there is an STR locus where the samples have a different genotype:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d9f5bcda",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_comparison[df_comparison[\"sample_difference\"] != 0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b78215d",
   "metadata": {},
   "source": [
    "### 4: Determining the relevance of observed STR variation\n",
    "\n",
    "The APC gene starts at position 112'702'498 on the forward strand of chromosome 5. You can find the entry for the APC gene in Ensembl [here](http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000134982;r=5:112707498-112846239;t=ENST00000257430). Using the start position of the APC gene, and the start and end positions of the mutated STR we detected, try to figure out if you expect the observed mutation to have an impact on the function of the APC gene.\n",
    "\n",
    "To make life easier, you can set the 'start' and 'end' variables in the code block below to the appropriate values. If you then run the code, it will print a link that you can follow to the region of interest. Look for the 'All phenotype-associated - short variants (SNPs and indels)' section of the page."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f632248c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: fill in start and end positions you are interested in\n",
    "start = \n",
    "end = \n",
    "\n",
    "print(f\"https://www.ensembl.org/Homo_sapiens/Location/View?db=core;r=5:{start}-{end};contigviewbottom=variation_set_ph_variants=labels\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb664d0b",
   "metadata": {},
   "source": [
    "After having explored the potential effects and phenotype associations on ensembl, do you expect our observed mutation to have an effect on the functionality of the *APC* gene product?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55e1d988",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Task: explain whether you expect the observed mutation to have a functional impact "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "218247c3",
   "metadata": {},
   "source": [
    "Which of the two samples do you expect to originate from the healthy sample? And which from the tumour sample?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6a282da8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Task: which sample came from healthy tissue, and which from tumour?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
