# Task: Exploring Genome Resources



### [The UCSC genome browser](https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr2%3A25160915%2D25168903&hgsid=1717435210_Ay0Jb5iAfpSxnoA7sDPbKV4hF3xA)

* originated from the Human Genome Project
* open source database
* most used general genome browser
* allows searching for specific genes
* reference genomes of many different species
* visualization of the genomes including many annotations
* implemented sequence alignment tool
* tools for analysis and downloading of the data

### [Human Genome Resources at NCBI](https://www.ncbi.nlm.nih.gov/genome/guide/human/)

* NCBI = National Center for Biotechnology Information
* human reference genomes which are part of the Human Genome Project
* serves as an entry point for genome reference data
  

### [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)

* information about genes and their clinical relevance
* record of reported relationships between human variants and health status
* phenotypic information about variants
* used in precision medicine and research

### [Ensembl](https://www.ensembl.org/index.html)

* genome browser for vertebrate genomes (many different species)
* information about sequence variations
* Perl API for direct access
* multiple alignment computation available
* provides a table of transcripts and according proteins encoded in these regions
* tools include
    * BLAST/BLAT: search genomes for specific sequence
    * BioMart: data-mining tool to export datasets
    * VEP (variant effect predictor): analyse variants and predict functional consequences
    * File chameleon: convert files

### [DECIPHER](https://www.deciphergenomics.org/browser)

* short for "Database of genomic variation and phenotype in humans using Ensembl resources"
* helps associating genomic variants to a phenotype
* used for clinical diagnosis

### [COSMIC](https://cancer.sanger.ac.uk/cosmic/browse/genome)

* short for "Catalogue Of Somatic Mutations In Cancer"
* largest and most comprehensive resource regarding somatic mutations in human cancer
* genome-wide screen data
* high precision data

### [dbSNP]()

* short for "Database of Short Genetic Variations"
* includes SNPs (single nucleotide variants) and small-scale insertions and deletions
* population-specific data
* neutral variations and clinical mutations
