# Task: Exploring Genome Resources



### The UCSC genome browser

* originated from the Human Genome Project
* open source database
* most used general genome browser
* allows searching for specific genes
* reference genomes of many different species
* visualization of the genomes including many annotations
* implemented sequence alignment tool
* tools for analysis and downloading of the data

### Human Genome Resources at NCBI

* NCBI = National Center for Biotechnology Information
* human reference genomes which are part of the Human Genome Project
* serves as an entry point for genome reference data
  

### ClinVar

* information about genes and their clinical relevance
* record of reported relationships between human variants and health status
* phenotypic information about variants
* used in precision medicine and research

### Ensembl

* genome browser for vertebrate genomes (many different species)
* information about sequence variations
* Perl API for direct access
* multiple alignment computation available
* provides a table of transcripts and according proteins encoded in these regions
* tools include
    * BLAST/BLAT: search genomes for specific sequence
    * BioMart: data-mining tool to export datasets
    * VEP (variant effect predictor): analyse variants and predict functional consequences
    * File chameleon: convert files

### DECIPHER

* short for "Database of genomic variation and phenotype in humans using Ensembl resources"
* helps associating genomic variants to a phenotype
* used for clinical diagnosis

### COSMIC

* short for "Catalogue Of Somatic Mutations In Cancer"
* largest and most comprehensive resource regarding somatic mutations in human cancer
* genome-wide screen data
* high precision data

### dbSNP

* short for "Database of Short Genetic Variations"
* includes SNPs (single nucleotide variants) and small-scale insertions and deletions
* population-specific data
* neutral variations and clinical mutations
