
#  Task: Estimate Storage Requirements for 1000 Genomes

WGS also called "Whole genome sequencing" is the process, by which 
you try to get the DNA nucleotide sequence of the whole genome, which includes the whole chromosomal DNA and mitochondria. 
One of the first approaches was the capillary sequencing method. With this method, it was possible to nearly sequence the whole genome but 
the problem was that it was too expensive and inefficient for commercial purposes. 
Over time, the nanopore-sequencing technology became more prevalent because it is much quicker and can sequence 
overall longer sequences. The only disadvantage is that it is less accurate. 
Methods for analyzing sequencing data are leading to a better understanding of diseases and target their prevention.
The costs for a WGS range from $1,000 to $5,000 per genome. 


Whole exome sequencing, short WES is the process for the identification of coding regions of genes responsible for protein function. 
The disadvantage here is there are no alterations seen in the introns, which could have an effect on the genome dynamic. 
Overall the advantage is that mutations are detected faster and allow to characterize the specific diseases.
The costs for a WES range from $500 to $2,000 per sample. 

#  SAM 

SAM stands for "Sequence Alignment/Map". It is a text-based file format used in bioinformatics and genomics to store information
about the alignment of a sequence to a reference genome. 

# BAM (Binary Alignment Map) 

BAM is the compressed binary version of a SAM file and has the specific benefit of finding overlapping locations quickly.
Its format allows a very compact saving and is faster to read and write compared to its text-based counterpart, the SAM file.
Bam files are divided into a header and alignments. 
The header contains the sample name, sample length, and the alignment method. Meanwhile, the alignments part contains the read name,
sequence, quality, alignment information and custom tags. 
Chromosome, start coordinate, alignment quality and match descriptor string can be found under the read name. 

#  VCF

#  FASTA


